File: src/main/java/com/arcao/feedback/collector/AccountInfoCollector.java
package com.arcao.feedback.collector;

import android.content.Context;
import com.arcao.geocaching4locus.App;
import com.arcao.geocaching4locus.authentication.util.AccountManager;
import com.arcao.geocaching4locus.base.constants.PrefConstants;

public class AccountInfoCollector extends Collector {
	private final Context mContext;

	public AccountInfoCollector(Context context) {
		this.mContext = context.getApplicationContext();
	}

	@Override
	public String getName() {
		return "AccountInfo";
	}

	@Override
	protected String collect() {
		final StringBuilder sb = new StringBuilder();

		AccountManager helper = App.get(mContext).getAccountManager();

		if (!helper.hasAccount()) {
			sb.append("No Account").append("\n");
		} else {
			//noinspection ConstantConditions
			sb.append("NAME=").append(helper.getAccount().name).append("\n");

			sb.append("\n--- RESTRICTIONS ---\n");
			sb.append(new SharedPreferencesCollector(mContext, PrefConstants.RESTRICTION_STORAGE_NAME).collect());
		}
		return sb.toString();
	}
}

File: app/src/main/java/com/arcao/feedback/collector/AccountInfoCollector.kt
package com.arcao.feedback.collector

import com.arcao.geocaching4locus.App
import com.arcao.geocaching4locus.base.constants.PrefConstants
import com.arcao.geocaching4locus.data.account.AccountManager

class AccountInfoCollector(private val app: App, private val accountManager: AccountManager) : Collector() {
    override val name: String
        get() = "AccountInfo"

    override suspend fun collect(): String {
        val sb = StringBuilder()

        val account = accountManager.account
        if (account == null) {
            sb.append("No Account").append("\n")
        } else {
            sb.append("NAME=").append(account.userName).append("\n")

            sb.append("\n--- RESTRICTIONS ---\n")
            sb.append(SharedPreferencesCollector(app, PrefConstants.RESTRICTION_STORAGE_NAME).collect())
        }
        return sb.toString()
    }
}
